test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/AbstractLocalClusterFactory.java

```diff
@@ -443,6 +443,18 @@ private void copyExtraConfigFiles() {
             });
         }
 
+        public void writeToKeystoreFile() {
+            final Path keystoreFile = workingDir.resolve("config").resolve("elasticsearch.keystore");
+            try {
+                Files.deleteIfExists(keystoreFile);
```

Comment:
This is pretty heavy-handed but it helps avoid the complexity if dealing with settings removal.

Reply:
My main concern is whether this is reflective of what users will actually do? I presume it's more likely they will update the keystore in-place, rather than delete and recreate.

Reply:
What about when a secrets file is used? Does reloading secrets not work in that case or does it only support a keystore? If not, we may actually want to throw an error in that case.

Reply:
Oh ya, good question -- the reload settings API only applies to the keystore, and that's what I was concerned with here. AFAICT, the secrets-file based path could also benefit from the ability to trigger writing to the file. Essentially, changes to the secrets file are tracked automatically so a useful test flow could be:

1. update some secret value in a mutable settings provider that's tracked by the secrets file
2. call "write"
3. check that the changes were applied (no need for calling reload)

I'm not 100% confident though, and don't have a concrete test use case in mind for the above.

Is it alright if I just throw an unsupported op exception here if `usesSecureSecretsFile` is true and leave it for a future follow up (I can own it)? I'm not sure right now if there are additional nuances the secrets-file-based implementation but kinda blocked by not having the ability to re-write the keystore file.

Reply:
Oh ya, good question -- the reload secure settings API only applies to the keystore, that's what I was concerned with here. The API is not available in Serverless (since we don't use the keystore there in the first place). 

AFAICT, the secrets-file based path could also benefit from the ability to trigger flushing new secure settings to file. Essentially, changes to the secrets file are tracked automatically so a useful test flow could be:

1. update some secret value in a mutable settings provider that's tracked by the secrets file
2. call "write"
3. check that the changes were applied

I'm not 100% confident though, and don't have a concrete test use case in mind for the above.

Is it alright if I just throw an unsupported op exception here if `usesSecureSecretsFile` is true and leave it for a future follow up (I can own it)? I'm not sure right now if there are additional nuances the secrets-file-based implementation but kinda blocked by not having the ability to re-write the keystore file.

This means that any test calling `updateStoredSecureSettings` cannot run in Serverless. I'm running a `test-serverless-update` but the JWT test suite or any other suite I'm aiming to use this in (cross cluster stuff) should not be supported in Serverless.

Reply:
Perfect, I've pushed the change. I can open a Jira to track handling this for the secrets file in the future.